src/output/variables/precipitation.jl

```diff
@@ -92,6 +92,105 @@ Fields are: $(TYPEDFIELDS)"""
     transform::F = (x) -> 1000x     # [m] to [mm]
 end
 
+--------------
+
+"""Defines netCDF output for a specific variables, see [`VorticityOutput`](@ref) for details.
+Fields are: $(TYPEDFIELDS)"""
+@kwdef mutable struct ConvectiveSnowOutput{F, R} <: AbstractOutputVariable
+    name::String = "snow_conv"
+    unit::String = "mm"
+    long_name::String = "accumulated convective snow"
+    dims_xyzt::NTuple{4, Bool} = (true, true, false, true)
+    missing_value::Float64 = NaN
+    compression_level::Int = 3
+    shuffle::Bool = true
+    keepbits::Int = 20
+    transform::F = (x) -> 1000x                     # [m] to [mm]
+    rate::R = ConvectiveSnowRateOutput()   # include here to be called at finalize!
+end
+
+path(::ConvectiveSnowOutput, simulation) =
+    simulation.diagnostic_variables.physics.snow_convection
+
+# at finalize step postprocess the convective snow to get the rate
+finalize!(output::NetCDFOutput, variable::ConvectiveSnowOutput, args...) = output!(output, variable.rate, variable)
+
+abstract type AbstractSnowRateOutputVariable <: AbstractOutputVariable end
+
+"""Defines netCDF output for a specific variables, see [`VorticityOutput`](@ref) for details.
+Fields are: $(TYPEDFIELDS)"""
+@kwdef mutable struct ConvectiveSnowRateOutput{F} <: AbstractRainRateOutputVariable
+    name::String = "snow_conv_rate"
+    unit::String = "mm/hr"
+    long_name::String = "convective snow rate"
+    dims_xyzt::NTuple{4, Bool} = (true, true, false, true)
+    missing_value::Float64 = NaN
+    compression_level::Int = 3
+    shuffle::Bool = true
+    keepbits::Int = 7
+    transform::F = (x) -> 1000x     # [m] to [mm]
+end
+
+function output!(
+    output::NetCDFOutput,
+    variable::AbstractSnowRateOutputVariable,
+    acc_variable::AbstractOutputVariable,
+)
```

Comment:
This is exactly the same as for rain precipitation no? Then we can just make this reuse that code and we should call things `AbstractRateOutputVariable` for consistency, whether its snow or rain, it doesn't matter.